src/vs/workbench/browser/parts/editor/editorParts.ts

```diff
@@ -127,9 +127,13 @@ export class EditorParts extends MultiWindowParts<EditorPart> implements IEditor
 			this.updateGlobalContextKeys();
 			this._onDidActiveGroupChange.fire(group);
 		}));
-		disposables.add(part.onDidAddGroup(group => this._onDidAddGroup.fire(group)));
+		disposables.add(part.onDidAddGroup(group => {
+			this.registerRunGroupContextKeyProviders(group);
+			this._onDidAddGroup.fire(group);
+		}));
 		disposables.add(part.onDidRemoveGroup(group => {
 			this.removeGroupScopedContextKeys(group);
+			this.unregisterRunGroupContextKeyProviders(group);
 			this._onDidRemoveGroup.fire(group);
 		}));
 		disposables.add(part.onDidMoveGroup(group => this._onDidMoveGroup.fire(group)));
```

Comment:
This feels like the wrong location, I would add these listeners to the own `registerListeners` method and use the events from the class. The intent here is just to delegate to the outside.